[expo-notifications] 55.0.11 iOS build fails: type 'EXPermissionsService' has no member 'parsePermission'

### Minimal reproducible example

https://github.com/weitude/expo-notifications-parse-permission-repro

### Steps to reproduce

1. Clone the repro repo (uses expo-notifications@55.0.11)
2. Run `eas build -p ios`
3. Build fails at Xcode compilation with:

   PermissionsModule.swift:42:49: error: type 'EXPermissionsService' has no member 'parsePermission'
     promise.resolver(EXPermissionsService.parsePermission(fromRequester: permission))

   The Swift code in PermissionsModule.swift (introduced in 55.0.11) calls
   EXPermissionsService.parsePermission(fromRequester:), which maps to the ObjC method
   +[EXPermissionsService parsePermissionFromRequester:]. However, this method is only
   implemented in EXPermissionsService.m — it is NOT declared in the public header
   (EXPermissionsService.h in expo-modules-core), so Swift cannot see it.

4. Downgrading to expo-notifications@55.0.10 fixes the build (this version does not
   have the parsePermission call in PermissionsModule.swift).

Platform: iOS
Environment: EAS Build (cloud)
Package manager: npm (also reproducible with pnpm)

### Environment

```text
expo-env-info 2.0.11 environment info:
    System:
      OS: macOS 26.3
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 22.14.0
      npm: 11.4.1
      Watchman: 2026.03.02.00
    Managers:
      CocoaPods: 1.16.2
    SDKs:
      iOS SDK:
        Platforms: DriverKit 25.2, iOS 26.2, macOS 26.2, tvOS 26.2, visionOS 26.2, watchOS 26.2
    IDEs:
      Xcode: 26.3/17C529
    npmPackages:
      expo: ^55.0.0 => 55.0.4
      react: 19.2.0 => 19.2.0
      react-native: 0.83.2 => 0.83.2
    npmGlobalPackages:
      eas-cli: 18.1.0
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
Running 17 checks on your project...
  17/17 checks passed. No issues detected!
```

--- a/packages/expo-modules-core/ios/Legacy/Services/Permissions/EXPermissionsService.h
+++ b/packages/expo-modules-core/ios/Legacy/Services/Permissions/EXPermissionsService.h
@@ -9,8 +9,6 @@
 
 + (NSString *)permissionStringForStatus:(EXPermissionStatus)status;
 
-+ (NSDictionary *)parsePermissionFromRequester:(NSDictionary *)permission;
-
 - (void)askForGlobalPermissionUsingRequesterClass:(Class)requesterClass
                                     withResolver:(EXPromiseResolveBlock)resolver
                                     withRejecter:(EXPromiseRejectBlock)reject;

--- a/packages/expo-notifications/CHANGELOG.md
+++ b/packages/expo-notifications/CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- [ios] Fixed build issue with Expo Go in `PermissionsModule.swift` ([#43690](https://github.com/expo/expo/pull/43690) by [@vonovak](https://github.com/vonovak))
+
 ### 💡 Others
 
 ## 55.0.11 — 2026-03-05

--- a/packages/expo-notifications/ios/ExpoNotifications/Permissions/PermissionsModule.swift
+++ b/packages/expo-notifications/ios/ExpoNotifications/Permissions/PermissionsModule.swift
@@ -36,10 +36,17 @@ public class PermissionsModule: Module {
       // Call `requestAuthorization` directly to ensure new options are always
       // forwarded to the OS, even if notifications were previously granted.
       // iOS safely handles repeated calls to `requestAuthorization(options:)`.
-      // Expo Go notifications permissions are not scoped
+      // Expo Go notifications permissions are not scoped.
+      // We use a selector on SDK 55 because Expo Go doesn't have `parsePermissionFromRequester` in the public header.
       let resolver: EXPromiseResolveBlock = { result in
         if let permission = result as? [AnyHashable: Any] {
-          promise.resolver(EXPermissionsService.parsePermission(fromRequester: permission))
+          let selector = NSSelectorFromString("parsePermissionFromRequester:")
+          if EXPermissionsService.responds(to: selector),
+            let parsed = EXPermissionsService.perform(selector, with: permission)?.takeUnretainedValue() as? [AnyHashable: Any] {
+            promise.resolver(parsed)
+          } else {
+            promise.legacyRejecter("ERR_PERMISSIONS_REQUEST_NOTIFICATIONS", "Unexpected error in requestPermissionsAsync: EXPermissionsService doesn't respond to parsePermissionFromRequester:", nil)
+          }
         } else {
           promise.legacyRejecter("ERR_PERMISSIONS_REQUEST_NOTIFICATIONS", "Unexpected permission result type", nil)
         }

